plugins/nodes/src/js/filters/NodesRegionFilter.ts

```diff
@@ -0,0 +1,21 @@
+import DSLFilterTypes from "#SRC/js/constants/DSLFilterTypes";
+import DSLFilter from "#SRC/js/structs/DSLFilter";
+
+/**
+ * This filter handles the `region:XXXX` for nodes
+ */
+const filter: DSLFilter = {
+  /*
+   * Handle all `region:XXXX` attribute filters that we can handle.
+   */
+  filterCanHandle: (filterType, filterArguments) =>
+    filterType === DSLFilterTypes.ATTRIB && filterArguments.label === "region",
```

Comment:
My brain is having a hard time understanding how we are able to drop the `regions.includes`... checks in these functions?

Reply:
My brain is having a hard time understanding why the `regions.includes` was present. 😄

When the `DSLUtil` checks which filters to apply it asks every filters `filterCanHandle`. Our filter now says: "i can handle things with a region-label". It formerly said: "i can handle things with a region-label **and** a region-value that is within all the region-values that occur in the nodes table".

We now got all the filters that claim that they can handle the current `DSLExpression` and call `filterApply` on them. We now do a "show if the region matches `node.getRegionName().toLowerCase()`". We formerly did a "show if the region matches `node.getRegionName().toLowerCase() ` AND that region is an element of all the regions in the `NodesTable`" (which it must be anyway if a node matches on that).

I hope that makes things clearer. If not i think i can try again with more code.

Reply:
Makes sense - thanks!! 🙌